Reason: The OpenSSL_add_all_ciphers api doesn't was removed in OpenSSL 1.1
so we just replace the check with something that exists in OpenSSL and LibreSSL

---
--- configure	2012-09-17 13:30:07.000000000 +0200
+++ -	2021-02-03 16:20:59.395787656 +0100
@@ -4878,9 +4878,9 @@
 
 done
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for OpenSSL_add_all_ciphers in -lcrypto" >&5
-$as_echo_n "checking for OpenSSL_add_all_ciphers in -lcrypto... " >&6; }
-if ${ac_cv_lib_crypto_OpenSSL_add_all_ciphers+:} false; then :
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for OPENSSL_init_crypto in -lcrypto" >&5
+$as_echo_n "checking for OPENSSL_init_crypto in -lcrypto... " >&6; }
+if ${ac_cv_lib_crypto_OPENSSL_init_crypto+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   ac_check_lib_save_LIBS=$LIBS
@@ -4894,27 +4894,27 @@
 #ifdef __cplusplus
 extern "C"
 #endif
-char OpenSSL_add_all_ciphers ();
+char OPENSSL_init_crypto ();
 int
 main ()
 {
-return OpenSSL_add_all_ciphers ();
+return OPENSSL_init_crypto ();
   ;
   return 0;
 }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"; then :
-  ac_cv_lib_crypto_OpenSSL_add_all_ciphers=yes
+  ac_cv_lib_crypto_OPENSSL_init_crypto=yes
 else
-  ac_cv_lib_crypto_OpenSSL_add_all_ciphers=no
+  ac_cv_lib_crypto_OPENSSL_init_crypto=no
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_crypto_OpenSSL_add_all_ciphers" >&5
-$as_echo "$ac_cv_lib_crypto_OpenSSL_add_all_ciphers" >&6; }
-if test "x$ac_cv_lib_crypto_OpenSSL_add_all_ciphers" = xyes; then :
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_crypto_OPENSSL_init_crypto" >&5
+$as_echo "$ac_cv_lib_crypto_OPENSSL_init_crypto" >&6; }
+if test "x$ac_cv_lib_crypto_OPENSSL_init_crypto" = xyes; then :
   cat >>confdefs.h <<_ACEOF
 #define HAVE_LIBCRYPTO 1
 _ACEOF
